### PR TITLE
Add extensions to build separate artifacts with MKL and/or CUDA enabled

### LINKED ISSUE #4: Current default build is MKL
This is painful for machines which don't have MKL installed. The MKL version bound against is the one using Intel's OpenMP, which is even less likely to be available on a machine (plus it's MKL-ML, not regular MKL nor MKL DNN - now DNNL).

We should make the build system a little more configurable, or at least set the default build to be something that works everywhere (even if the performance is worse).

I keep running into this when testing things on different machines here, none of which have MKL, and I have to get approvals for new inbound libraries.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux
- TensorFlow installed from (source or binary): source
- TensorFlow version: 2.0

### PR DESCRIPTION
**Pull request by saudet**
Fixes #4

Usage should be clear from the modifications to the README.md. I've tested this on Linux, all builds pass and it runs, but some of the unit tests fail with CUDA. I think it just has to do with limited memory on my GPU though.

### CODE REVIEW

File: tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/c_api/AbstractTF_Tensor.java [hunk 1]
```diff
@@ -32,14 +32,7 @@
...
         @Override public void call(Pointer data, long len, Pointer arg) { }
-    };
-
-    static {
-        PointerScope s = PointerScope.getInnerScope();
-        if (s != null) {
-            s.detach(dummyDeallocator);
-        }
-    }
+    }.retainReference();
```
Comment (Craigacp) [inline tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/c_api/AbstractTF_Tensor.java]: This change doesn't look related to the rest of the PR?

Comment (saudet) [reply tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/c_api/AbstractTF_Tensor.java]: It's needed for JavaCPP 1.5.2, which changed a bit how `PointerScope` works to accommodate for reference counting. It doesn't affect anything since it's not being used right now, and it might never be either, if that gets fixed upstream and this hack is no longer needed.

**Comment by karllessard** [comment]
This seems good to me, @sjamesr do you also want to take a look? Basically it allows us to distribute TF in 4 flavors: "vanilla", "w/mkl", "w/gpu" and "w/mkl+gpu"
